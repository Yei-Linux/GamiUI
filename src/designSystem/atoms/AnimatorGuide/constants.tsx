export const Guiders: any = {
  inlove: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "0px", y: "0px" },
    dimensions: { width: "95px", height: "80px" },
  },
  smart: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "-100px", y: "0px" },
    dimensions: { width: "95px", height: "80px" },
  },
  worried: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "95px", y: "0px" },
    dimensions: { width: "95px", height: "80px" },
  },
  thinking: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "0px", y: "-95px" },
    dimensions: { width: "95px", height: "80px" },
  },
  yawning: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "-100px", y: "-95px" },
    dimensions: { width: "95px", height: "80px" },
  },
  drunk: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "95px", y: "-95px" },
    dimensions: { width: "95px", height: "80px" },
  },
  surprised: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "0px", y: "-190px" },
    dimensions: { width: "95px", height: "80px" },
  },
  upset: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "-100px", y: "-190px" },
    dimensions: { width: "95px", height: "80px" },
  },
  sad: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "95px", y: "-190px" },
    dimensions: { width: "95px", height: "80px" },
  },
  chill: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "0px", y: "0px" },
    dimensions: { width: "82px", height: "100px" },
  },
  strong: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "-85px", y: "0px" },
    dimensions: { width: "110px", height: "100px" },
  },
  box: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "95px", y: "0px" },
    dimensions: { width: "95px", height: "100px" },
  },
  byciclying: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "0px", y: "-100px" },
    dimensions: { width: "90px", height: "110px" },
  },
  basquet: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "-95px", y: "-102px" },
    dimensions: { width: "95px", height: "90px" },
  },
  uleule: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "95px", y: "-102px" },
    dimensions: { width: "95px", height: "90px" },
  },
  run: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "0px", y: "-215px" },
    dimensions: { width: "100px", height: "90px" },
  },
  skater: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "-100px", y: "-195px" },
    dimensions: { width: "95px", height: "100px" },
  },
  jumper: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "95px", y: "-195px" },
    dimensions: { width: "95px", height: "100px" },
  },
  singer: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "-100px", y: "-330px" },
    dimensions: { width: "110px", height: "120px" },
  },
};

export const options = {
  type: Object.keys(Guiders),
};
