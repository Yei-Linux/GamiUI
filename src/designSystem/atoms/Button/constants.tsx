export const options = {
  type: [
    "primary",
    "secondary",
    "tertiary",
    "cuaternary",
    "text",
    "link",
    "success",
    "warning",
    "danger",
    "info",
  ] as const,
};
