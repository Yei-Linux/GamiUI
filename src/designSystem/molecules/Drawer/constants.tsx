export const drawerTranstionByStates = {
  mask: {
    open: {
      from: {
        duration: 1.5,
        opacity: 0.3,
      },
      to: {
        opacity: 1,
      },
    },
    close: {
      from: null,
      to: {
        duration: 0.5,
        opacity: 0.01,
      },
    },
  },
  drawer: {
    open: {
      to: {
        duration: 0.8,
        delay: 0,
        position: { axis: 'xPercent', value: 100 },
      },
    },
    close: {
      to: {
        duration: 0.8,
        delay: 0,
        position: { axis: 'xPercent', value: -100 },
      },
    },
  },
}

export const options = {
  type: [] as const,
}
