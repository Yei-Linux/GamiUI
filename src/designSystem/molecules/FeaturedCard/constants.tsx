export const options = {
  type: ['sky', 'skin', 'earth', 'spring'] as const,
}
