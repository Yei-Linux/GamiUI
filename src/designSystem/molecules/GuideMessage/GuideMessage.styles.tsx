import styled from "@emotion/styled";
import tw from "twin.macro";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const GuideMessageWrapper = styled.div<{
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  direction: "left" | "right";
}>`
  ${tw`flex justify-center items-center`}

  ${(props: any) => twinStyles(props)};
  flex-direction: ${(props) =>
    props.direction == "left" ? "row" : "row-reverse"};
`;
