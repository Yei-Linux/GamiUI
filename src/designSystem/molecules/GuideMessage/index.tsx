export { default } from "./GuideMessage";
export { options } from "./constants";
