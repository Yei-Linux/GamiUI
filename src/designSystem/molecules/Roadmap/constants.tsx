export const RoadMapTypes: any = {
  primary:
    'https://cdn.dribbble.com/users/989466/screenshots/6645205/canopy-two-dots-rpg-map-drib-crop-desert-_2x_4x.png?compress=1&resize=1600x1200',
  secondary:
    'https://cdn.dribbble.com/users/121405/screenshots/6644930/canopy-two-dots-rpg-map-drib-crop-swamp-_2x_4x.png?compress=1&resize=1600x1200',
  tertiary:
    'https://cdn.dribbble.com/users/989466/screenshots/6639761/canopy-two-dots-rpg-map-drib-crop-volcanic-_2x_4x.png?compress=1&resize=1600x1200',
  cuaternary:
    'https://cdn.dribbble.com/users/121405/screenshots/14668817/media/1a4b4bcb8b0be19480c56395214eb052.png?compress=1&resize=1600x1200',
  pentanary:
    'https://cdn.dribbble.com/users/121405/screenshots/6639386/canopy-two-dots-rpg-map-drib-crop-ice-_2x_4x.png?compress=1&resize=1600x1200',
}

export const options = {
  names: Object.keys(RoadMapTypes),
}
