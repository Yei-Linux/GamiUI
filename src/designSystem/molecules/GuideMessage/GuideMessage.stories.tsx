import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";
import { options as optionsGuide } from "./constants";

import GuideMessage from ".";

const Template = getTemplate(GuideMessage);

export default {
  title: "Molecules/GuideMessage",
  component: GuideMessage,
  args: { ...GuideMessage.defaultProps },
  argTypes: {
    message: { control: { type: "text" } },
    guideType: { control: { type: "select", options: optionsGuide.type } },
    ...stylesControl,
  },
} as Meta;

export const Left = Template.bind({});
Left.args = {
  guideType: "run",
  message: "Left Direction",
};

export const Right = Template.bind({});
Right.args = {
  guideType: "skater",
  direction: "right",
  message: "Right Direction",
};
