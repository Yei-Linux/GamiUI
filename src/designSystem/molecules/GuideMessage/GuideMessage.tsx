import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { GuidesNames } from "../../../core/domain/types";
import AnimatorGuide from "../../atoms/AnimatorGuide/AnimatorGuide";
import Message from "../../atoms/Message/Message";
import { GuideMessageWrapper } from "./GuideMessage.styles";

export interface GuideMessageProps extends IGeneralProps {
  /**
   * Guide Type
   */
  guideType: GuidesNames;
  /**
   * Message Text
   */
  message: string;
  /**
   * Message Direction
   */
  direction: "left" | "right";
  /**
   * Background
   */
  background: string;
}

const GuideMessage = ({
  guideType,
  message,
  direction,
  background,
  ...args
}: GuideMessageProps) => (
  <GuideMessageWrapper direction={direction} {...args}>
    <AnimatorGuide type={guideType} />
    <Message
      style={{ marginTop: "15px" }}
      text={message}
      direction={direction}
      background={background}
    />
  </GuideMessageWrapper>
);

GuideMessage.defaultProps = {
  direction: "left",
  guideType: "singer",
  background: "#7868e6",
};

export default GuideMessage;
