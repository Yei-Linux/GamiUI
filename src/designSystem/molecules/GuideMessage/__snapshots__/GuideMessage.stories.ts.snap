// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/GuideMessage Left 1`] = `
<div
  className="css-1qr63or-GuideMessageWrapper ewha4wa0"
  direction="left"
>
  <div
    className="css-zh1rva-AnimatorGuideImg e9zrumb0"
  />
  <div
    className="css-16p81sx-MessageWrapper e6z3dsf0"
    direction="left"
    style={
      Object {
        "marginTop": "15px",
      }
    }
  >
    Left Direction
  </div>
</div>
`;

exports[`Storyshots Molecules/GuideMessage Right 1`] = `
<div
  className="css-v95ugg-GuideMessageWrapper ewha4wa0"
  direction="right"
>
  <div
    className="css-u6s7mk-AnimatorGuideImg e9zrumb0"
  />
  <div
    className="css-j0rdok-MessageWrapper e6z3dsf0"
    direction="right"
    style={
      Object {
        "marginTop": "15px",
      }
    }
  >
    Right Direction
  </div>
</div>
`;
